process/block/baseProcessHeaderV3_test.go

```diff
@@ -0,0 +1,765 @@
+						},
+					}
+				},
+			},
+		}
+
+		body := &block.Body{
+			MiniBlocks: []*block.MiniBlock{
+				{
+					ReceiverShardID: bp.shardCoordinator.SelfId() + 1,
+					SenderShardID:   bp.shardCoordinator.SelfId(),
+				},
+			},
+		}
+		_, _, err := bp.createMiniBlockHeaderHandlers(body, nil, true)
+		require.NoError(t, err)
+		require.True(t, wasPutCalled)
+	})
+}
+
+func TestBaseProcessor_cacheIntermediateTxsForHeader(t *testing.T) {
+	t.Parallel()
+
+	t.Run("marshal error", func(t *testing.T) {
+		t.Parallel()
+
+		bp := &baseProcessor{
+			marshalizer: &marshallerMock.MarshalizerStub{
+				MarshalCalled: func(obj interface{}) ([]byte, error) {
+					return nil, errExpected
+				},
+			},
+			txCoordinator: &testscommon.TransactionCoordinatorMock{},
+		}
+
+		err := bp.cacheIntermediateTxsForHeader(headerHash)
+		require.Equal(t, errExpected, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		wasPutCalled := false
+		bp := &baseProcessor{
+			marshalizer:   &marshallerMock.MarshalizerStub{},
+			txCoordinator: &testscommon.TransactionCoordinatorMock{},
+			dataPool: &dataRetrieverMock.PoolsHolderStub{
+				PostProcessTransactionsCalled: func() storage.Cacher {
+					return &cache.CacherStub{
+						PutCalled: func(key []byte, value interface{}, sizeInBytes int) (evicted bool) {
+							wasPutCalled = true
+							return false
+						},
+					}
+				},
+			},
+		}
+
+		err := bp.cacheIntermediateTxsForHeader(headerHash)
+		require.NoError(t, err)
+		require.True(t, wasPutCalled)
+	})
+}
+
+func TestBaseProcessor_saveExecutedData(t *testing.T) {
+	t.Parallel()
+
+	t.Run("not header v3", func(t *testing.T) {
+		t.Parallel()
+
+		bp := &baseProcessor{}
+		header := &testscommon.HeaderHandlerStub{
+			IsHeaderV3Called: func() bool {
+				return false
+			},
+		}
+
+		err := bp.saveExecutedData(header, headerHash)
+		require.Nil(t, err)
+	})
+	t.Run("header v3 with no execution results", func(t *testing.T) {
+		t.Parallel()
+
+		bp := &baseProcessor{
+			txCoordinator: &testscommon.TransactionCoordinatorMock{
+				GetAllIntermediateTxsCalled: func() map[block.Type]map[string]data.TransactionHandler {
+					return make(map[block.Type]map[string]data.TransactionHandler)
+				},
+			},
+			marshalizer: &marshallerMock.MarshalizerStub{},
+			dataPool: &dataRetrieverMock.PoolsHolderStub{
+				PostProcessTransactionsCalled: func() storage.Cacher {
+					return &cache.CacherStub{
+						GetCalled: func(key []byte) (value interface{}, ok bool) {
+							return []byte("marshalled map"), true
+						},
+					}
+				},
+			},
+		}
+		header := &testscommon.HeaderHandlerStub{
+			IsHeaderV3Called: func() bool {
+				return true
+			},
+			GetExecutionResultsHandlersCalled: func() []data.BaseExecutionResultHandler {
+				return []data.BaseExecutionResultHandler{}
+			},
+		}
+
+		err := bp.saveExecutedData(header, headerHash)
+		require.NoError(t, err)
+	})
+	t.Run("saveMiniBlocksFromExecutionResults path", func(t *testing.T) {
```

Comment:
not a strong opinion, but i think it might be better to have a separate main test for these flow, instead of keeping them in one big main test